afl-fuzz.c

```diff
@@ -5117,7 +5118,7 @@ static u8 fuzz_one(char** argv) {
      this entry ourselves (was_fuzzed), or if it has gone through deterministic
      testing in earlier, resumed runs (passed_det). */
 
-  if (skip_deterministic || queue_cur->was_fuzzed || queue_cur->passed_det)
+  if ((skip_deterministic && !use_dictionary) || queue_cur->was_fuzzed || queue_cur->passed_det)
     goto havoc_stage;
```

Comment:
would it be simpler to leave this condition, and `goto dict_stage` here, and at
```
dict_stage:
  if (!use_dictionary) goto havoc_stage;
```
?